sw/device/silicon_creator/lib/ownership/owner_block.c

```diff
@@ -0,0 +1,175 @@
+    }
+    remain -= item->length;
+    offset += item->length;
+    switch (item->tag) {
+      case kTlvTagApplicationKey:
+        if (keyring->length < ARRAYSIZE(keyring->key)) {
+          keyring->key[keyring->length++] =
+              (const owner_application_key_t *)item;
+        }
+        break;
+      case kTlvTagFlashConfig:
+        if (config->flash)
+          return kErrorOwnershipDuplicateItem;
+        config->flash = (const owner_flash_config_t *)item;
+        break;
+      case kTlvTagInfoConfig:
+        if (config->info)
+          return kErrorOwnershipDuplicateItem;
+        config->info = (const owner_flash_info_config_t *)item;
+        break;
+      case kTlvTagRescueConfig:
+        if (config->rescue)
+          return kErrorOwnershipDuplicateItem;
+        config->rescue = (const owner_rescue_config_t *)item;
+        break;
+      default:
+        /* unrecognized item. skip. */
+        config->unknown_tags += 1;
+    }
+  }
+  return kErrorOk;
+}
+
+#define MUBI(ex_) (ex_) ? kMultiBitBool4True : kMultiBitBool4False
+
+rom_error_t owner_block_flash_apply(const owner_flash_config_t *flash,
+                                    uint32_t config_side,
+                                    uint32_t primary_side) {
+  if (!flash)
+    return kErrorOk;
+  uint32_t start = config_side == kBootDataSlotA   ? 0
+                   : config_side == kBootDataSlotB ? kFlashBankSize
+                                                   : 0xFFFFFFFF;
+  uint32_t end = config_side == kBootDataSlotA   ? kFlashBankSize
+                 : config_side == kBootDataSlotB ? 2 * kFlashBankSize
+                                                 : 0;
+  size_t len = (flash->header.length - sizeof(owner_flash_config_t)) /
+               sizeof(owner_flash_region_t);
+  if (len >= 8) {
+    return kErrorOwnershipFlashConfigLenth;
+  }
+
+  const owner_flash_region_t *config = flash->config;
+  for (size_t i = 0; i < len; ++i, ++config) {
+    if (config->start >= start && config->start + config->size <= end) {
+      flash_ctrl_cfg_t cfg = {
+          .scrambling = MUBI(config->properties & kOwnerFlashPropertyScramble),
+          .ecc = MUBI(config->properties & kOwnerFlashPropertyEcc),
+          .he = MUBI(config->properties & kOwnerFlashPropertyHighEndurance),
+      };
+      flash_ctrl_perms_t perm = {
+          .read = MUBI(config->properties & kOwnerFlashPropertyRead),
+          .write = MUBI(config->properties & kOwnerFlashPropertyProgram),
+          .erase = MUBI(config->properties & kOwnerFlashPropertyErase),
+      };
+      if (config_side == primary_side &&
+          config->properties & kOwnerFlashPropertyProtectWhenPrimary) {
+        perm.write = kMultiBitBool4False;
+        perm.erase = kMultiBitBool4False;
+      }
+      flash_ctrl_data_region_protect(i, config->start, config->size, perm, cfg);
+    }
+  }
+  return kErrorOk;
+}
+
+static inline bool is_owner_page(const owner_info_page_t *config) {
```

Comment:
`hardened_bool_t` here to avoid comparisons to zero in the calling side.

Reply:
Done.